sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -112,7 +112,11 @@ private static void BuildCanonicalizedHeaders(StringBuilder stringBuilder, HttpM
                 }
             }
 
-            headers.Sort(static (x, y) => string.CompareOrdinal(x.Name, y.Name));
+            headers.Sort(static (x, y) =>
```

Comment:
TODO: Make a sample app with .NET framework 4.6.2 to determine the relative order of strings with this algorithm.

- Create List of strings, each string is a single character.  Each character that is legal in HTTP header names should appear once in the list - https://developers.cloudflare.com/rules/transform/request-header-modification/reference/header-format/
- Sort via this algorithm
- Implement .NET platform-agnostic version of this algorithm with a Dictionary<Character, Integer> where the keys are the HTTP header name-legal characters, and the values are their values generated from the sort above.

Reply:
It appears this was resolved with:
```
            headers.Sort(static (x, y) =>
            {
                CultureInfo cultureInfo = new CultureInfo(Constants.EnUsCulture, useUserOverride: false);
                return string.Compare(x.Name, y.Name, cultureInfo, CompareOptions.IgnoreSymbols);
            });
```